src/UpdatesPublisher.js

```diff
@@ -0,0 +1,76 @@
+const {createProducer} = require('windbreaker-service-util/queue')
+const ChangesStream = require('changes-stream')
+const Promise = require('bluebird')
+
+class UpdatesPublisher {
+  constructor (options) {
+    this.producerOptions = options.producerOptions
+    this.amqUrl = options.amqUrl
+    this.registryUrl = options.registryUrl
+    this.logger = options.logger
+  }
+
+  async setupProducer () {
+    const {logger, amqUrl, producerOptions} = this
+    try {
+      this.producer = await createProducer({logger, amqUrl, producerOptions})
+      logger.info('producer successfully created')
+    } catch (error) {
+      logger.error(error)
+      logger.info('error creating producer, retrying ...')
```

Comment:
This would be a bit better later on when we have some sort of logging system in place

```
logger.error('Error creating producer', error)
logger.info(`Attempting to reinitialize queue producer...`)
```